File: gsonpath-compiler/src/main/java/gsonpath/generator/adapter/AutoGsonAdapterProperties.java
package gsonpath.generator.adapter;

import com.google.gson.FieldNamingPolicy;
import gsonpath.GsonFieldValidationType;
import gsonpath.PathSubstitution;

class AutoGsonAdapterProperties {
    final boolean fieldsRequireAnnotation;
    final char flattenDelimiter;
    final boolean serializeNulls;
    final String rootField;
    final GsonFieldValidationType gsonFieldValidationType;
    final FieldNamingPolicy gsonFieldNamingPolicy;
    final PathSubstitution[] pathSubstitutions;

    AutoGsonAdapterProperties(boolean fieldsRequireAnnotation, char flattenDelimiter, boolean serializeNulls,
                              String rootField, GsonFieldValidationType gsonFieldValidationType,
                              FieldNamingPolicy gsonFieldNamingPolicy, PathSubstitution[] pathSubstitutions) {

        this.fieldsRequireAnnotation = fieldsRequireAnnotation;
        this.flattenDelimiter = flattenDelimiter;
        this.serializeNulls = serializeNulls;
        this.rootField = rootField;
        this.gsonFieldValidationType = gsonFieldValidationType;
        this.gsonFieldNamingPolicy = gsonFieldNamingPolicy;
        this.pathSubstitutions = pathSubstitutions;
    }
}

File: gsonpath-compiler/src/main/java/gsonpath/generator/adapter/properties/AutoGsonAdapterProperties.kt
package gsonpath.generator.adapter.properties

import com.google.gson.FieldNamingPolicy
import gsonpath.GsonFieldValidationType
import gsonpath.PathSubstitution

class AutoGsonAdapterProperties(
        val fieldsRequireAnnotation: Boolean,
        val flattenDelimiter: Char,
        val serializeNulls: Boolean,
        val rootField: String,
        val gsonFieldValidationType: GsonFieldValidationType,
        val gsonFieldNamingPolicy: FieldNamingPolicy,
        val pathSubstitutions: Array<PathSubstitution>)
